projects/deja-js/component/editor/deja-editor.component.ts

```diff
@@ -368,19 +371,21 @@ export class DejaEditorComponent extends Destroy implements OnChanges, AfterView
             this.instance.focus = focus;
             return;
         }
-        const range = this.instance.getSelection().getRanges(true)[0];
-        if (!range) {
-            this.instance.insertHtml(replace);
-            return;
-        }
-        const text = this.firstTextNode(range);
-        if (text) {
-            this.replaceWord(text, replace);
-        } else {
-            this.instance.insertHtml(replace);
+        if (this.instance) {
```

Comment:
A la première ligne de la fonction        
 if (!replace) {
    return;
 }

On pourrait mettre 
 if (!replace || !this.instance) {
    return;
 }

Ce serait plus simple non?

Mais la question que je me pose est comment replace a été appelé avec instance qui est nulle?

Reply:
Je reprends cette PR, c'est moi qui en suis à l'origine.
J'ai une erreur qui revient régulièrement sur le `getSelectedText()` mais je ne sais pas si elle vient de `hasActiveSelection`ou de `replace` dans le doute j'ai préféré ajouter un controle dans le replace au retour du  `getSelectedText()`
Je peux déplacer ce contrôle comme tu le préconises, c'est comme tu veux

Reply:
Oui, déplace le test, et montre moi ou tu fais appelle à cette méthode dans ton code.